brave/brave-core
# Issue 3419: Mitigate HSTS fingerprinting

fixes https://github.com/brave/brave-browser/issues/3419
## Description

HSTS super-cookies are used as a fingerprinting vector.  This change disallow's third parties from setting trackable security headers like "Strict-Transport-Security", "Expect-CT", "Public-Key-Pins", "Public-Key-Pins-Report-Only"

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

1. Navigate to `https://avatars2.githubusercontent.com` and disable `Connections Encrypted`
2. Navigate to `https://jsfiddle.net` and disable `Connections Encrypted` (see: https://github.com/brave/brave-browser/issues/3424)
3. Clear Browsing Data 
4. Navigate to `https://jsfiddle.net/x5mqb807/` and verify that `Connections Encrypted` is disabled
5. Verify that content from `avatars2.githubusercontent.com` loads with a 301 redirect
6. Navigate to ` http://avatars2.githubusercontent.com`, reload ` http://avatars2.githubusercontent.com` (not https) and verify it loads with a 307 redirect
7. Reload `https://jsfiddle.net/x5mqb807/` in the first tab and verify the image is loaded with a 307 redirect.

Before navigating to `avatars2.githubusercontent.com`:

<img width="1199" alt="screen shot 2019-02-20 at 2 07 33 pm" src="https://user-images.githubusercontent.com/1903815/53128175-16885a00-3519-11e9-85cb-65f1c55bc245.png">


After navigating to  `avatars2.githubusercontent.com`:

<img width="1189" alt="screen shot 2019-02-20 at 2 04 04 pm" src="https://user-images.githubusercontent.com/1903815/53128029-b1ccff80-3518-11e9-8b36-8706b793c46f.png">


## Reviewer Checklist:
